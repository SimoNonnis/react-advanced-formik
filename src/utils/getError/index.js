function getError(touched, errors) {
  return function returnError(key) {
    return touched[key] && errors[key];
  };
}

export default getError;
