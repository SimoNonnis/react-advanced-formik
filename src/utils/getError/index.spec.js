/* eslint-env jest */
import getError from './';

const mockTouched = {
  one: true,
  two: false,
};

const mockErrors = {
  one: 'one',
  two: 'two',
};

const mockValue = {
  one: 'one',
};

describe('Test getError()', () => {
  it('should return a function waiting for a key', () => {
    expect(getError(mockTouched, mockErrors)).toMatchSnapshot();
  });
});

describe('Test getError()', () => {
  it('should get the error', () => {
    const showError = getError(mockTouched, mockErrors);
    expect(showError(mockValue.one)).toMatchSnapshot();
  });
});
