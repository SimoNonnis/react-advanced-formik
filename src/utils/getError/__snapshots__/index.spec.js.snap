// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test getError() should get the error 1`] = `"one"`;

exports[`Test getError() should return a function waiting for a key 1`] = `[Function]`;
